### DISCUSSION TITLE
Developer-centric surveys on formatter / linter usage?

### DISCUSSION THREAD

**Discussion by marinegor** [astral-sh/ruff#13370] [General]
Hi everyone,

I wonder if there's any demographics on how many people use certain linters/formatters? I know some project-centric surveys (like [this one](https://www.linkedin.com/pulse/finding-best-linter-python-complete-analysis-alon-raizman/)), but here I can't reason about particular people. Does the community have any data, perhaps?

Some background: we're trying to introduce `ruff` to a big codebase, but doing formatting in one PR and adding it to `.git-blame-ignore-revs` isn't an option (there were problems with black's formatting style before and people are worried that ruff would bring the same, but at the same time the need for a formatter is high, so we're trying to weight all pros and cons).

**Comment by MichaReiser** [comment]
> I wonder if there's any demographics on how many people use certain linters/formatters? I know some project-centric surveys (like [this one](https://www.linkedin.com/pulse/finding-best-linter-python-complete-analysis-alon-raizman/)), but here I can't reason about particular people. Does the community have any data, perhaps?

Is there something specific you're looking for? You can use [PyPI's download statistics](https://pypistats.org/packages/ruff) to a rough usage comparison.

Another alternative is to use github search and search for `ruff format` or the existence of `tool.ruff` in pyproject.tomls

> Some background: we're trying to introduce ruff to a big codebase, but doing formatting in one PR and adding it to .git-blame-ignore-revs isn't an option (there were problems with black's formatting style before and people are worried that ruff would bring the same, but at the same time the need for a formatter is high, so we're trying to weight all pros and cons).

I don't think I understand this part. I can see that `.git-blame-ignore-revs` doesn't help evaluate a style guide. It's only to hide the commit from git-blame once you committed to a tool.

> and people are worried that ruff would bring the same,

Ruff's style guide is inspired by Black's. So, there's a high chance that Ruff formats the code identically. Feel free to share examples of formatting that you think could be improved.

**Comment by marinegor** [comment]
>Is there something specific you're looking for?

Something like "x% of developers use ruff, y% use black, ...", where x+y+... don't necessarily add up to 100% (because one person might use different formatters in different projects). 


>I don't think I understand this part

I mean, we don't want to blindly do one commit with a big automatic formatting, and want to evaluate all changes in smaller PRs to ensure the readability "isn't worse" (not my words). But since manual labor == lots of work, we want some extra pros for using ruff, and one of such points could be "80% of developers use it to assist themselves in writing code / finding silly bugs, hence we should enable it by default as fast as possible".

**Reply by zanieb** [reply]
There's this [StackOverflow Survey](https://survey.stackoverflow.co/2024/technology#2-other-frameworks-and-libraries) which showed that 84% of those that use Ruff want to continue using it.
